Add feature to navigate and display files within the current working directory.

ntdkhiem opened the pull request.

ntdkhiem committed:
Display contents onto the menu

ntdkhiem committed:
Display its content if the target is a file.

ntdkhiem committed:
Merge branch 'main' of https://github.com/CupOfGeo/adaptable-antelopes into files_explore

BoraxTheClean commented on scroll_menu.py:
```diff
@@ -7,45 +11,88 @@
+    def exit_(event: str) -> None:
+        get_app().exit()
+
+    kb.add("tab")(focus_next)
+    kb.add("s-tab")(focus_previous)
```
Should we also add arrow keys here (and maybe even vim bindings)?

BoraxTheClean replied:
We should also have a keyboard shortcut that opens a file, like enter.

BoraxTheClean commented on scroll_menu.py:
```diff
@@ -7,45 +11,88 @@
+                0, Window(content=FormattedTextControl(file_content))
+            )
+        elif isdir(join(target_dir, target_content)):
+            pass
```
I'm running this locally and clicking on a directory doesn't do anything.

It seems like that is intentional, should we add some placeholder functionality and add a Issue to implement that?

BoraxTheClean replied:
From @CupOfGeo: We also need an ⬆️ move up on directory button for this use case.

BoraxTheClean commented on scroll_menu.py:
```diff
@@ -56,62 +103,13 @@ def main() -> None:
-    kb.add("tab")(focus_next)
-    kb.add("s-tab")(focus_previous)
-
+    kb = setup_key_bindings()
     # Create and run application.
     application = Application(
         layout=layout, key_bindings=kb, full_screen=True, mouse_support=True
     )
     application.run()
```
Since we're not going to be running `scroll_menu.py` locally to invoke the application, this should be hooked up to the text_editor code.

BoraxTheClean commented on scroll_menu.py:
```diff
@@ -7,45 +11,88 @@
+            Frame(
+                Button(
+                    text=item,
+                    handler=functools.partial(self.display_content, item, "."),
```
If `"."` is the directory path parameter in `self.display_content`, can we make that  a constant?

BoraxTheClean commented on scroll_menu.py:
```diff
@@ -7,45 +11,88 @@
+
+
+class ScrollableMenu:
+    """Main class for scrollable menu"""
```
```suggestion
    """
    Scrollable Menu UI Element.
    
    Displays contents of the a given working directory and allows users to inspect files. 
    In the future users can drill down into directories.
    """
```
Does this description look good to you? You can commit it from the PR UI.

BoraxTheClean commented on scroll_menu.py:
```diff
@@ -7,45 +11,88 @@
+                Label(text="File's content here"),
+                Frame(ScrollablePane(HSplit(children=frames))),
+            ],
+            padding_char="|",
+            padding=1,
```
Can we make these a constant?

BoraxTheClean commented on scroll_menu.py:
```diff
@@ -7,45 +11,88 @@
+        """Display content.
+
+        If target's content is a file, display it in the left column.
+        If the target_content is a directory, replace the right column with its content.
```
This is a TODO
```suggestion
        TODO: If the target_content is a directory, replace the right column with its content.
```

BoraxTheClean reviewed (changes requested):
We should remove the files in the `notes/`dir.

ntdkhiem committed:
Update docstring for ScrollableMenu

Co-authored-by: Owen Collier-Ridge <owen.collier.ridge@gmail.com>

ntdkhiem committed:
TODO for displaying contents in a given directory

Co-authored-by: Owen Collier-Ridge <owen.collier.ridge@gmail.com>

ntdkhiem committed:
Remove dummy folder

ntdkhiem committed:
Update .gitignore to ignore .vim

ntdkhiem committed:
Merged main branch

ntdkhiem committed:
Replace with new scrollable menu

:heavy_plus_sign: Able to display contents from given directory
:hammer: TODO: Need to focus the cursor back to the dialog after display contents from given directory.

ntdkhiem committed:
Merge branch 'files_explore' of https://github.com/CupOfGeo/adaptable-antelopes into files_explore

ntdkhiem committed:
Create __init__.py

For importing constants from constants.py

ntdkhiem committed:
:hammer: Refocus the cursor

Refocus the cursor back to the scroll dialog.

ntdkhiem committed:
Delete scroll_menu.py

CupOfGeo commented on src/constants.py:
```diff
@@ -0,0 +1,4 @@
+# For scroll/scroll_menu.py
+CURRENT_WORK_DIR = "."
+PADDING_CHAR = "|"
+PADDING_WIDTH = 1
```
I don't know if you need a whole file for this you can just add them to the top of scroll menu for readability and more flexible to change

ntdkhiem committed:
Clarify each step when display contents from dir.

ntdkhiem committed:
Add a move-up one directory button

ntdkhiem committed:
Add cancel button

ntdkhiem committed:
Remove line numbers

ntdkhiem committed:
Cover the space on the left column when open the scroll menu

Also add a hint to style the menu when needed.

ntdkhiem committed:
Add title at the bottom bar.

ntdkhiem committed:
Display file's name as the title when open.
